File: explori/src/Models/Collection.js
/**
Collection.js

Creates a class called Collection

It is a collection of items

**/

//util lib
import _ from "lodash";

//root model
import Model from "./Model";

//Import extra models
import Item from "./Item";

class Collection extends Model {
  //initialize with the name and Items of the collection
  constructor(id, source, name, items) {
    super({
      id,
      source,
      name,
      items
    });
  }

  static parse(raw_collection) {
    let pre_parse = _.assign(raw_collection, {
      items: Item.parseList(raw_collection.items)
    });

    return Model.parse(pre_parse, Collection);
  }

  static parseList(raw_collections) {
    console.log(raw_collections);

    let pre_parse = _.map(raw_collections, function(raw_collection) {
      return _.assign(raw_collection, {
        items: Item.parseList(raw_collection.items)
      });
    });

    return Model.parseList(pre_parse, Collection);
  }
}

export default Collection;

File: scraper/models/Collection.py
#
# ./models/Collection.py
# Collection Model
#
# A collection has a name and items
#

import json

class Collection:

    # collection constructor
    def __init__(self, name, items):
        self.name = name
        self.items = items

    # convert model to dictionary
    def __dict__(self):

        item_dicts = []
        for item in self.items:
            item_dicts.append(item.__dict__())

        return {
            "name": self.name,
            "items": item_dicts
        }

    # return json version of object
    def __json__(self):
        json_str = json.dumps(self.__dict__())
        return json_str

    # string version of the Collection model
    def __str__(self):
        return self.__json__()
